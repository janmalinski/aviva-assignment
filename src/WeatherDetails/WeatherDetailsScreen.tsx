import React, { useCallback } from 'react';
import { StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import WeatherDetails from '../components/WeatherDetails';
import { NavigationProp, RouteProp, useNavigation, useRoute } from '@react-navigation/native';
import { TRootNavigatorParams } from '../navigation/RootNavigator';
import Button from '../components/Button';

const WeatherDetailsScreen = () => {
  const route = useRoute<RouteProp<TRootNavigatorParams, "WeatherDetails">>();
  const navigation = useNavigation<NavigationProp<TRootNavigatorParams, "WeatherDetails">>();

  const navigateToHome = useCallback(() => {
    navigation.navigate("Home");
  }, [navigation]);

  return (
    <SafeAreaView style={styles.container} edges={['top']}>
      <WeatherDetails screen='WeatherDetails' weatherData={{...route.params}}  />
      <Button title='Back' onPress={navigateToHome}/>
    </SafeAreaView>
  )
}

export default WeatherDetailsScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 40,
    backgroundColor: '#e4f6f4'
  },
})