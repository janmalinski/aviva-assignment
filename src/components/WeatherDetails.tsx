import React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';
import { Entypo } from '@expo/vector-icons'; 

import { TRootNavigatorParams, TWeatherDetails } from '../navigation/RootNavigator';
import parseFloatToOneDecimal from '../helpers/ParseFloatToOneDecimal';
import capitalizeFirstLetter from '../helpers/CapitalizeFirstLetter';

interface IWeatherDetails {
    weatherData: TWeatherDetails;
    screen: keyof TRootNavigatorParams;
} 

const WeatherDetails = ({ weatherData, screen} : IWeatherDetails) => {
  const {city, description, temp, temp_max, temp_min, feels_like, humidity, pressure, icon} = weatherData;
  return (
    <View testID='weather-details' style={{ marginTop: screen === 'Home' ? '65%' : 0}}>
        {screen === 'Home' ?
          <View style={styles.homeContainer}>
            <View style={styles.row}>
              <Entypo name="location-pin" size={24} color="black" />
              <Text style={styles.textXl}>{city}</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.currentTemperatureText}>{parseFloatToOneDecimal(temp)}</Text><Text style={styles.currentTemperatureSymbol}>˚</Text>
            </View>
            <Image style={styles.icon} source={{uri: `http://openweathermap.org/img/wn/${icon}@2x.png`}} />
          </View>
          :  screen === 'WeatherDetails' && 
          <View style={styles.weatherDetailsContainer}>
            <View style={[styles.row, styles.gapWithAlignmentCenter]}>
              <Entypo name="location-pin" size={24} color="black" />
              <Text style={styles.textXl}>{city}</Text>
            </View>
            <View style={styles.row}>
              <Image style={[styles.icon, styles.marginLeftMinusTwenty]} source={{uri: `http://openweathermap.org/img/wn/${icon}@2x.png`}} />
              <Text style={[styles.textXl, styles.temperatureText]}>{parseFloatToOneDecimal(temp)}˚</Text>
              <View style={styles.verticallDivider} />
              <Text style={[styles.text, styles.wrapText]}>{capitalizeFirstLetter(description)}.</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.text}>Maximum temprature: {parseFloatToOneDecimal(temp_max)}˚</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.text}>Minimum temprature: {parseFloatToOneDecimal(temp_min)}˚</Text>
            </View>
            <View style={styles.row}>
              <Text style={styles.text}>Feels like: {parseFloatToOneDecimal(feels_like)}˚</Text>
            </View>
            <Text style={styles.text}>Humidity: {humidity}%</Text>
            <Text style={styles.text}>Pressure: {pressure}mb</Text>
          </View>
      }
    </View> 
  )
}

export default WeatherDetails;

const styles = StyleSheet.create({
  homeContainer: {
    alignItems: 'center'
  },
  textXl: {
    fontSize: 22,
    fontFamily: 'Actor-Regular',
  },
  icon: {
    height: 100,
    width: 100,
    marginTop: -15
  },
  marginLeftMinusTwenty: {
    marginLeft: -20
  },
  currentTemperatureText: {
    fontSize: 42,
    fontFamily: 'Actor-Regular',
    alignSelf: 'center',
  },
  currentTemperatureSymbol: {
    marginLeft: 5,
    marginTop: 7,
    fontSize: 38,
  },
  temperatureSymbol: {
    marginTop: 5,
    fontSize: 16,
  },
  row: {
    flexDirection: 'row',
  },
  gapWithAlignmentCenter: {
    gap: 10,
    alignItems: 'center'
  },
  text: {
    fontSize: 18,
    fontFamily: 'Actor-Regular',
    lineHeight: 24
  },
  temperatureText: {
    marginTop: 5,
    marginLeft: 5,
  },
  wrapText: {
    marginTop: 5,
    flex: 1,
    flexWrap: 'wrap'
  },
  weatherDetailsContainer: {
    marginBottom: 40,
  },
  verticallDivider: {
    height: '80%',
    width: 1,
    backgroundColor: '#000',
    marginHorizontal: 10
  }
})