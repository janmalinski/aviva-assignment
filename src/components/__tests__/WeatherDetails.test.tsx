import React from 'react';
import { render } from '@testing-library/react-native';

import WeatherDetails from '../WeatherDetails';

const screen = 'Home';
const weatherData = {
    city: "Pozna≈Ñ",
    description: "light intensity shower rain",
    feels_like: 14,
    humidity: 79,
    icon: "09d",
    pressure: 1003,
    temp: 14.44,
    temp_max: 16.83,
    temp_min: 14.12
};

const WeatherDetailsEl = (
    <WeatherDetails screen={screen}  weatherData={weatherData}/>
);


describe('Rendering WeatherDetails component', () => {

    it('Should match a snapshot', () => {
        const elJson = render(WeatherDetailsEl).toJSON();
        expect(elJson).toMatchSnapshot(); 
    });
});

