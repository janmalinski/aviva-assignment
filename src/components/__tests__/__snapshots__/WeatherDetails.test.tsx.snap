// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering WeatherDetails component Should match a snapshot 1`] = `
<View
  style={
    {
      "marginTop": "65%",
    }
  }
  testID="weather-details"
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          {
            "fontFamily": "Actor-Regular",
            "fontSize": 22,
          }
        }
      >
        Poznań
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "fontFamily": "Actor-Regular",
            "fontSize": 42,
          }
        }
      >
        14.4
      </Text>
      <Text
        style={
          {
            "fontSize": 38,
            "marginLeft": 5,
            "marginTop": 7,
          }
        }
      >
        ˚
      </Text>
    </View>
    <Image
      source={
        {
          "uri": "http://openweathermap.org/img/wn/09d@2x.png",
        }
      }
      style={
        {
          "height": 100,
          "marginTop": -15,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
